src/wayland/shm/mod.rs

```diff
@@ -154,6 +156,13 @@ impl ShmState {
     pub fn global(&self) -> GlobalId {
         self.shm.clone()
     }
+
+    /// Updates the list of formats advertised by the global.
+    ///
+    /// This will only affect new binds to the wl_shm global.
+    pub fn add_formats(&mut self, formats: impl IntoIterator<Item = wl_shm::Format>) {
```

Comment:
I guess this begs the question, do we need a way to remove formats from future binds?

Reply:
Given that we would still need to support the old formats for already bound clients, I just see this function as a way to lazily initialize the list (until a renderer is created) not to further alter the list later.